transport/grpc/outbound.go

```diff
@@ -217,15 +218,15 @@ func (o *Outbound) invoke(
 			responseBody,
 			callOptions...,
 		),
-	)
+		yarpcerrors.FromError(err).Code())
 	if err != nil {
 		return invokeErrorToYARPCError(err, *responseMD)
 	}
 	// Service name match validation, return yarpcerrors.CodeInternal error if not match
 	if match, resSvcName := checkServiceMatch(request.Service, *responseMD); !match {
 		// If service doesn't match => we got response => span must not be nil
 		return transport.UpdateSpanWithErr(span, yarpcerrors.InternalErrorf("service name sent from the request "+
-			"does not match the service name received in the response: sent %q, got: %q", request.Service, resSvcName))
+			"does not match the service name received in the response: sent %q, got: %q", request.Service, resSvcName), yarpcerrors.FromError(err).Code())
```

Comment:
What's the purpose of it in here? Shouldn't it be just yarpcerrors.CodeInternal?